ngraph/frontend/onnx_editor/include/onnx_editor/editor.hpp

```diff
@@ -110,11 +111,57 @@ namespace ngraph
             /// \param out_file_path A path to the file where the modified model should be dumped.
             void serialize(const std::string& out_file_path) const;
 
+            /// \brief Returns the InputEdge based on a node (node name or output name)
+            ///        and an input (input name or input index).
+            ///
+            /// \note  The node name can be ambiguous (many ONNX nodes can have the same name).
+            ///        In such a case the algorthim tries to match the given node name
+            ///        with the input name (providing an input index is not enough).
+            ///        If a unique edge is found, it will be returned.
+            ///        If InputEdge cannot be determined based on parameter values an ngraph_error
+            ///        exception will be thrown.
+            ///
+            /// \param node A node helper structure created based on a node name
+            ///             or a node output name.
+            ///
+            /// \param input An input helper structure created based on a input name
+            ///              or a input index.
+            InputEdge find_input_edge(const EditorNode& node, const EditorInput& input);
+
+            /// \brief Returns an OutputEdge based on a node (node name or output name)
+            ///        and an output (output name or output index).
+            ///
+            /// \note  The node name can be ambiguous (many ONNX nodes can have the same name).
+            ///        In such a case the algorthim will try to match the given node name
+            ///        with the output name (providing an output index is not enough).
+            ///        If after such operation a found edge is unique, it is returned.
+            ///        If OutputEdge cannot be determined based on given params the ngraph_error
+            ///        exception is thrown.
+            ///
+            /// \param node A node helper structure created based on a node name
+            ///             or a node output name.
+            ///
+            /// \param output A output helper structure created based on a output name
+            ///               or a output index.
+            OutputEdge find_output_edge(const EditorNode& node, const EditorOutput& output);
+
+            /// \brief Returns an OutputEdge based on a output name.
+            ///
+            /// \note  The output name guarantees the uniqueness of the edge.
+            ///
+            /// \param output_name A node output name.
+            ///
+            OutputEdge find_output_edge(const std::string& output_name);
```

Comment:
Method that gives an identification for a tensor is missing. A tensor is something that identifies data in the graph that can play only the input role or only the output role. It is something that user can see in Netron when browsing an ONNX model. Using existing methods I have always specify the role: input or output. But FE API can represent the place in the graph without this specialization. One of the valid queries in this case is just check whether a given name represent a tensor in a given graph. With the methods from these PR I have always separately check it for inputs and outputs that complicates code and I trend to postpone such query to the later stage keeping potentially incorrect objects referring to some place in the graph.

Before the PR I've added method `validate_tensor_name` which gives `bool` reporting whether such tensor exists regardless of specialization. I would like to see such methods or a way to represent tensors without specialization.

Reply:
If I understand you correctly, you mean about adding to the editor interface method:
```
bool is_correct_tensor_name(const std::string& tensor_name)
```
Am I right?
A tensor can be also identified by output name. Do we need also sth like:
 ```
bool is_correct_output_name(const std::string& output_name)
```
?

Reply:
Ideally I want to get something that represents a valid tensor by its name. Probably name is enough, but it depends on how much of the context around that name we would like to capture in such ID to provide fast enough queries for the neighbors (producing nodes, consuming node etc.). But `is_correct_output_name` looks enough for this particular task of checking whether we have a tensor with a given name. Output name == tensor name for ONNX.

Reply:
To make this method more compatible with the current API, my proposition is `bool is_correct_and_unambiguous_node(const EditorNode& node) const` which allows to use both `output_name` and and `node_name` - https://github.com/openvinotoolkit/openvino/pull/4927/commits/3ffdb48edb87a78c70359b44b60e39eef1018de9